django/db/models/fields/__init__.py

```diff
@@ -703,6 +704,9 @@ def contribute_to_class(self, cls, name, private_only=False, virtual_only=NOT_PR
             setattr(cls, 'get_%s_display' % self.name,
                     curry(cls._get_FIELD_display, field=self))
 
+        if self.readonly:
+            cls._meta.has_readonly_fields = True
```

Comment:
I think this would be better as a `cached_property` on `Options` that iterates all the fields and returns True if any are readonly - it would be a bit less surprising to me, as then it's defined entirely within the `Options` class